src/test/java/org/candlepin/resource/EnvironmentResourceTest.java

```diff
@@ -0,0 +1,217 @@
+    @Mock
+    private RdbmsExceptionTranslator rdbmsExceptionTranslator;
+    @Mock
+    private JobManager jobManager;
+    @Mock
+    private DTOValidator validator;
+    @Mock
+    private ContentAccessManager contentAccessManager;
+    @Mock
+    private CertificateSerialCurator certificateSerialCurator;
+    private I18n i18n;
+    private ModelTranslator translator;
+
+    private EnvironmentResource environmentResource;
+
+    private Owner owner;
+    private Environment environment1;
+
+    @BeforeEach
+    void setUp() {
+        this.i18n = I18nFactory.getI18n(getClass(), Locale.US, I18nFactory.FALLBACK);
+        this.translator = new SimpleModelTranslator();
+        this.translator.registerTranslator(new EnvironmentTranslator(),
+            Environment.class, EnvironmentDTO.class);
+        this.translator.registerTranslator(new ContentTranslator(), Content.class, ContentDTO.class);
+        this.translator.registerTranslator(new NestedOwnerTranslator(), Owner.class, NestedOwnerDTO.class);
+
+        this.environmentResource = new EnvironmentResource(
+            this.envCurator,
+            this.i18n,
+            this.envContentCurator,
+            this.consumerResource,
+            this.poolManager,
+            this.consumerCurator,
+            this.ownerContentCurator,
+            this.rdbmsExceptionTranslator,
+            this.translator,
+            this.jobManager,
+            this.validator,
+            this.contentAccessManager,
+            this.certificateSerialCurator
+        );
+
+        this.owner = new Owner("owner1", "Owner 1");
+        owner.setId("owner1");
+        this.environment1 = createEnvironment(owner, ENV_ID_1);
+    }
+
+    @Test
+    void environmentNotFound() {
+        assertThrows(NotFoundException.class, () -> this.environmentResource.getEnvironment(BAD_ID));
+    }
+
+    @Test
+    void environmentFound() {
+        when(this.envCurator.get(anyString())).thenReturn(this.environment1);
+
+        EnvironmentDTO environment = this.environmentResource.getEnvironment(ENV_ID_1);
+
+        assertNotNull(environment);
+    }
+
+    @Test
+    void nothingToDelete() {
+        assertThrows(NotFoundException.class, () -> this.environmentResource.deleteEnvironment(BAD_ID));
+    }
+
+    @Test
+    void canDeleteEmptyEnvironment() {
+        when(this.envCurator.get(anyString())).thenReturn(this.environment1);
+        CandlepinQuery<Consumer> mockedQuery = mockedQueryOf();
+        when(this.envCurator.getEnvironmentConsumers(eq(this.environment1)))
+            .thenReturn(mockedQuery);
+
+        this.environmentResource.deleteEnvironment(BAD_ID);
+
+        verifyZeroInteractions(this.poolManager);
+        verifyZeroInteractions(this.consumerCurator);
+        verifyZeroInteractions(this.certificateSerialCurator);
+        verify(this.envCurator).delete(eq(this.environment1));
+    }
+
+    @Test
+    void shouldCleanUpAfterDeletingEnvironment() {
```

Comment:
Great job adding unit tests, especially since there wasn't a unit test class for this resource, but let's also add a spec test too

Reply:
Done.